plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,427 @@
+package application_insights
+
+import (
+	"fmt"
+	"log"
+	"math"
+	"time"
+	"unsafe"
+
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type TelemetryTransmitter interface {
+	Track(appinsights.Telemetry)
+	Close() <-chan struct{}
+}
+
+type DiagnosticsMessageSubscriber interface {
+	Subscribe(appinsights.DiagnosticsMessageHandler) appinsights.DiagnosticsMessageListener
+}
+
+type ApplicationInsights struct {
+	InstrumentationKey      string
+	Timeout                 internal.Duration
+	EnableDiagnosticLogging bool
+	ContextTagSources       map[string]string
+	diagMsgSubscriber       DiagnosticsMessageSubscriber
+	transmitter             TelemetryTransmitter
+	diagMsgListener         appinsights.DiagnosticsMessageListener
+}
+
+var (
+	sampleConfig = `
+## Instrumentation key of the Application Insights resource.
+instrumentationKey = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
+
+## Timeout on close. If not provided, will default to 5s. 0s means no timeout (not recommended).
+# timeout = "5s"
+
+## Determines whether diagnostic logging (for Application Insights endpoint traffic) is enabled. Default is false.
+# enableDiagnosticLogging = "true"
+
+## ContextTagSources dictionary instructs the Application Insights plugin to set Application Insights context tags
```

Comment:
Add link to contextTagKeys
https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/Schema/PublicSchema/ContextTagKeys.bond

Reply:
OK